doc/designs/quic-design/record-layer.md

```diff
@@ -507,7 +507,7 @@ struct ossl_record_method_st {
      * multiple records in one go and buffer them.
      */
     int (*read_record)(OSSL_RECORD_LAYER *rl, void **rechandle, int *rversion,
-                      int *type, unsigned char **data, size_t *datalen,
+                      uint8_t *type, unsigned char **data, size_t *datalen,
```

Comment:
If we're going to change this type I think it has ripple effects in libssl, e.g. I would also suggest changing the type of `type` and `recvd_type` in `ssl3_read_bytes` and `dtls1_read_bytes`

Reply:
AFAIK these are assigned and not passed as pointer to them so type conversion should happen fine. But I can change it anyway if you prefer that.

Reply:
@mattcaswell it was quite spread throughout the code but AFAIK done now, please look.

Reply:
> it was quite spread throughout the code but AFAIK done now, please look.

Yeah - this is what I meant by a ripple effect.

> But I can change it anyway if you prefer that.

I really dislike this type dissonance where we use different types for the same thing in different places. Looks much better now.